### ISSUE TITLE
Gutenframe: Error previewing content

### ISSUE THREAD

**Issue by mmtr** [Automattic/wp-calypso#31790]
We have received a report where a user was not able to preview a post / page because of a `sameorigin` error when using Gutenframe (the previewed post / page was not allowing to be embed on the Calypso preview modal). 

<img width="989" alt="Screen Shot 2019-03-27 at 11 30 21" src="https://user-images.githubusercontent.com/1233880/55046122-bdfe2c80-5083-11e9-90c1-a97ffaef9804.png">

It was able to reproduce on the user site when SSP as the user, but somehow the issue automagically fixed itself and the previews work normally now.

The user site was unlaunched (and, therefore, private) but this seems to be unrelated to the issue (at least, I cannot reproduce it on my unlaunched test sites).

I'm opening this issue to see if we have any other report that can help us to identify the issue. If we don't have any, we'll close this issue.

Original report in p1553626629176200-slack-gutengroup-wpcomhappy.

**Comment by ersatzpenguin** [comment]
I have a related error on another unlaunched site that was reported in 1974560-zen, and discussed on p2EDhh-Kw-p2.

I am not so sure that this is related to Gutenframe specifically. Copying some of my notes:

I have been able to reproduce this behavior when logged in as the user, and trying to preview the post in Gutenframe. I have personally confirmed that it happens in latest version of Chrome, Firefox, and Safari.

However, the error goes away if you click on the “preview” link in the permalink section of the editor. When you come back from the permalink preview and click on the normal “preview” button, it loads just fine. That said, clearing the cache resets the issue and requires you to click the permalink preview to re-enable the correct function of the preview button in Gutenframe.

I have also investigated and found that this issue does not occur when you access the site via WP Admin and the classic editor—it simply pops out the preview in a new window as expected.

However, *the issue also exists using the WP Admin Block Editor.* When previewing from the WP Admin Block Editor pop-out, the preview just hangs forever on the “Generating Preview” page.

I am personally at a loss as to what the issue is actually related to, but it seems to be related to the Block Editor in some way since the classic editor doesn't have any trouble with it.
